oio/account/backend.py

```diff
@@ -202,83 +204,115 @@ def update_container(self, account_id, name, mtime, dtime, object_count,
                          bytes_used):
         conn = self.conn
         if not account_id or not name:
-            return None
-        _account_id = conn.hget('account:%s' % account_id, 'id')
-
-        if not _account_id:
-            if self.autocreate:
-                self.create_account(account_id)
-            else:
-                return None
-
-        lock = acquire_lock_with_timeout(
-                conn, AccountBackend.ckey(account_id, name), 1)
-        if not lock:
-            return None
-
-        data = conn.hgetall(AccountBackend.ckey(account_id, name))
-
-        record = {'name': name, 'mtime': mtime or dtime, 'dtime': dtime,
-                  'objects': object_count, 'bytes': bytes_used}
-        if data:
-            data['mtime'] = Timestamp(data['mtime'])
-            data['dtime'] = Timestamp(data['dtime'])
-
-            for r in ['name', 'mtime', 'dtime', 'objects', 'bytes']:
-                if record[r] is None and data[r] is not None:
-                    record[r] = data[r]
-            if data['mtime'] > record['mtime']:
-                record['mtime'] = data['mtime']
-            if data['dtime'] > record['dtime']:
-                record['dtime'] = data['dtime']
-
-        deleted = record['dtime'] >= record['mtime']
-
-        if not deleted:
-            incr_bytes_used = int_value(record.get('bytes'), 0) -\
-                int_value(data.get('bytes'), 0)
-            incr_object_count = int_value(record.get('objects'), 0) -\
-                int_value(data.get('objects'), 0)
-        elif record.get('mtime') > data.get('mtime'):
-            incr_bytes_used = - int_value(data.get('bytes'), 0)
-            incr_object_count = - int_value(data.get('objects'), 0)
-        else:
-            # The event has been delayed, the container has already
-            # been deleted, and the object and bytes statistics have
-            # already been reported to the account.
-            incr_bytes_used = 0
-            incr_object_count = 0
-
-        record.update({
-            'name': name,
-            'account_uid': account_id
-        })
-
-        # replace None values
-        for r in ['bytes', 'objects', 'mtime', 'dtime']:
-            if record[r] is None:
-                record[r] = 0
-
-        ct = {name: 0}
+            raise NotFound("Missing account or container")
+
+        if mtime is None:
+            mtime = '0'
+        if dtime is None:
+            dtime = '0'
+        if object_count is None:
+            object_count = 0
+        if bytes_used is None:
+            bytes_used = 0
+
+        lua = (
+               # With lua float we are losing precision for this reason
+               # we keep the number as a string
+               " local is_sup = function(a,b)\n"
```

Comment:
you should use
"""
script lua
"""
(but I don't check if conn.register_script accept string or only a tuple)